source/UIAUtils.py

```diff
@@ -203,3 +205,17 @@ def getValue(self,ID,ignoreMixedValues=False):
 			raise UIAMixedAttributeError
 		return val
 
+def shouldUseUIAConsole(setting=None):
+	"""Determines whether to use UIA in the Windows Console.
+@param setting: the config value to base this check on (if not provided,
+it is retrieved from config).
+	"""
+	if not setting:
+		setting = config.conf['UIA']['winConsoleImplementation']
+	if setting == "legacy":
+		return False
+	elif setting == "UIA":
+		return True
+	# #7497: Windows 10 Fall Creators Update has an incomplete UIA implementation for console windows, therefore for now we should ignore it.
+	# It does not implement caret/selection, and probably has no new text events.
+	return isWin10(1803)

```

Comment:
The docs for `isWin10` say `@note: this function will always return False for source copies of NVDA due to a Python bug.`

Is this true? if not, please update the docs for the `isWin10` function.

Reply:
It is True for python 2 and false for python 3.

Reply:
Yes, this is definitely at least true for Python 2.

Reply:
Ah, I think we are all on the same page, but I just realised my question "is it true" is quite open to interpretation. I was asking if the docs are correct. Please correct me if I'm mistaken with the following:
- Python 2: running from source causes the `isWin10` function to behave incorrectly, it always returns False.
- Python 3: the bug is fixed, `isWin10` function returns True when running from source on a Windows 10 machine.

If this makes it into Python 2 code (which is currently not the plan), we would need to look at this carefully to see if there is any way we can improve the situation, given how prevalent using the console is for people working from source copies. For now, no need to do anything, Bill could you add this to your checklist for the end of your project?